pageserver/src/tenant.rs

```diff
@@ -2159,6 +2272,8 @@ impl Tenant {
         // FIXME If the config file is not found, assume that we're attaching
         // a detached tenant and config is passed via attach command.
         // https://github.com/neondatabase/neon/issues/1555
+        // OR: we're loading after incomplete deletion that managed to remove config.
+        // Attention ^
```

Comment:
Was there a discussion you wanted to have about this? Unsure why we need to return default here, could just go with None which would seem to be more honest.

Wouldn't it move the handling to somewhere with more context? Or perhaps not. I think we currently create a tenant config for all tenants... And I think we can also receive the config via attach, so the above FIXME is a bit outdated anyways.

> OR: we're loading after incomplete deletion that managed to remove config.

Is this possible, mark file is removed *after* config?

Reply:
> Was there a discussion you wanted to have about this?

Sort of. To me this looks weird and indeed returning option would be more honest. From outside perspective when looking at `load_tenant_config` I'd imagine that it wont introduce a default one out of thin air. So I consider this as a hack, but one that appeared before this PR. This is why I decided to bring more attention to it. 

So for now lets leave it as is and it can be turned into option in a follow up. Objections?

> Is this possible, mark file is removed after config?

Yeah, this happens because mark file is removed last, so we can observe arbitrary state prior to mark file deletion

Reply:
Hmm okay to leave as is. `Attention ^' looks a bit weird and inactionable, but ... not as bad as likely outdated FIXME.

Yeah it is not clear what to do here.

> > Was there a discussion you wanted to have about this?
> 
> Sort of. To me this looks weird and indeed returning option would be more honest. From outside perspective when looking at `load_tenant_config` I'd imagine that it wont introduce a default one out of thin air. So I consider this as a hack, but one that appeared before this PR. This is why I decided to bring more attention to it.
> 
> So for now lets leave it as is and it can be turned into option in a follow up. Objections?
> 
> > Is this possible, mark file is removed after config?
> 
> Yeah, this happens because mark file is removed last, so we can observe arbitrary state prior to mark file deletion

I did not actually see this possibility but okay now I understand, we could have a crash in-between. Unsure if we have any chance of fixing this without a log and some atomicity. But unsure if I'll enjoy fixing a custom log either :) But it would be just a one place to fix.

Okay to resolve.

Reply:
Makes sense to remove `Attention ^`, will do.

> Unsure if we have any chance of fixing this without a log and some atomicity

Yeah, mark file is poor'mans log consisting of one entry :) Expect this to be improved under the scope of https://github.com/neondatabase/neon/issues/4636